File: app_movil/lib/features/models/registro_progreso.dart
/// Un registro de que el cliente completó una sesión de un ejercicio de su
/// rutina, con los valores reales que hizo (pueden diferir de lo planeado
/// en DetalleRutina). GET /rutinas/{id}/detalles/{id}/registros,
/// POST del mismo path.
class RegistroProgreso {
  final int id;
  final int detalleRutinaId;
  final DateTime fecha;
  final int seriesRealizadas;
  final int repeticionesRealizadas;
  final num pesoRealizado;
  final String? nota;

  const RegistroProgreso({
    required this.id,
    required this.detalleRutinaId,
    required this.fecha,
    required this.seriesRealizadas,
    required this.repeticionesRealizadas,
    required this.pesoRealizado,
    required this.nota,
  });

  factory RegistroProgreso.fromJson(Map<String, dynamic> json) => RegistroProgreso(
        id: json['id'] as int,
        detalleRutinaId: json['detalle_rutina_id'] as int,
        fecha: DateTime.parse(json['fecha'] as String),
        seriesRealizadas: json['series_realizadas'] as int,
        repeticionesRealizadas: json['repeticiones_realizadas'] as int,
        pesoRealizado: json['peso_realizado'] as num,
        nota: json['nota'] as String?,
      );
}

File: ms-rutinas/app/schemas/registro_progreso.py
from datetime import datetime

from pydantic import BaseModel, Field


class RegistroProgresoCreate(BaseModel):
    series_realizadas: int = Field(gt=0)
    repeticiones_realizadas: int = Field(gt=0)
    peso_realizado: float = Field(ge=0)
    nota: str | None = Field(default=None, max_length=500)


class RegistroProgresoResponse(BaseModel):
    id: int
    detalle_rutina_id: int
    fecha: datetime
    series_realizadas: int
    repeticiones_realizadas: int
    peso_realizado: float
    nota: str | None

    model_config = {"from_attributes": True}
